whois-update/src/main/java/net/ripe/db/whois/update/handler/transform/AsSetAttributeTransformer.java

```diff
@@ -0,0 +1,49 @@
+package net.ripe.db.whois.update.handler.transform;
+
+import com.google.common.collect.Maps;
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.RpslObjectBuilder;
+import net.ripe.db.whois.update.domain.Action;
+import net.ripe.db.whois.update.domain.Update;
+import net.ripe.db.whois.update.domain.UpdateContext;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Objects;
+
+@Component
+public class AsSetAttributeTransformer implements Transformer{
+
+    private final RpslObjectDao rpslObjectDao;
+
+    @Value("${whois.nonauth.source}")
+    private String nonAuthRipeSource;
+
+    public AsSetAttributeTransformer(final RpslObjectDao rpslObjectDao){
+        this.rpslObjectDao = rpslObjectDao;
+    }
+
+    @Override
+    public RpslObject transform(RpslObject rpslObject, Update update, UpdateContext updateContext, Action action) {
+        String asSetKey = rpslObject.getKey().toString();
```

Comment:
I would say return if action is delete or object type is not as-set

Reply:
Also if user has specified source as non-auth there is no need to perform a check? Return from here only?

Reply:
we need to check that the aut-num is non-auth as well. If both, aut-num and as-set has the same source I do a return now.